source/common/http/conn_manager_impl.h

```diff
@@ -128,10 +130,28 @@ class ConnectionManagerImpl : Logger::Loggable<Logger::Id::http>,
     Tracing::Span& activeSpan() override;
     Tracing::Config& tracingConfig() override;
 
+    // Functions to set or get iteration state.
+    bool canIterate() { return iteration_state_ == IterationState::Continue; }
+    bool shouldNotIterate() { return iteration_state_ != IterationState::Continue; }
+    bool stoppedAll() {
+      return iteration_state_ == IterationState::StopAllBuffer ||
+             iteration_state_ == IterationState::StopAllWatermark;
+    }
+    void allowIteration() { iteration_state_ = IterationState::Continue; }
```

Comment:
Can you ASSERT that iteration_state != Continue? (From my read of the code I think that is true?)

Reply:
Fixed. Thanks!